kitsune/users/views.py

```diff
@@ -326,13 +327,31 @@ def profile(request, template, username):
         raise Http404('No Profile matches the given query.')
 
     groups = user_profile.user.groups.all()
+
+    if user.profile.facebook:
+        # Get Facebook Username only from Facebook profile URL by using Facebook Graph API
+        # More about Facebook Graph API can be get from
+        # here: https://developers.facebook.com/docs/graph-api
+        facebook_graph_url = user.profile.facebook.replace(
+            'www.facebook.com', 'graph.facebook.com').replace(
+            'facebook.com', 'graph.facebook.com').replace(
+            'profile.php?id=', '')
```

Comment:
This string of replace calls is pretty brittle, and you have it duplicated in the Validator as well. Instead you should use a regular expression to extract the information you want from the string, and you should re-use that regular expression between this function and `FacebookValidator`.